src/app/modules/submission/email-submission/__tests__/email-submission.service.spec.ts

```diff
@@ -96,11 +100,13 @@ describe('email-submission.service', () => {
         new Set(),
       )
 
-      expect(emailData).toEqual({
-        jsonData: [],
-        autoReplyData: [generateSingleAnswerAutoreply(response)],
-        formData: [generateSingleAnswerFormData(response)],
-      })
+      expect(emailData).toEqual(
+        expect.objectContaining({
```

Comment:
for all these checks, would it better to separate the checks for each key of the object we expect? then if there's a bug in one of them, at least we can see the error in isolation

Reply:
edited